airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/tools.py

```diff
@@ -0,0 +1,47 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+import re
+from typing import Any, Mapping
+from urllib.parse import parse_qsl, urlparse
+
+import pendulum as pdm
+
+from .query import PARERNT_KEY
+
+
+class BulkTools:
+    @staticmethod
+    def camel_to_snake(camel_case: str):
+        snake_case = []
+        for char in camel_case:
+            if char.isupper():
+                snake_case.append("_" + char.lower())
+            else:
+                snake_case.append(char)
+        return "".join(snake_case).lstrip("_")
+
+    @staticmethod
+    def filename_from_url(job_result_url: str) -> str:
+        # Regular expression pattern to extract the filename
```

Comment:
can you add an example filename to help the reader?

Reply:
added